rxrabbit/src/main/java/com/meltwater/rxrabbit/impl/SingleChannelConsumer.java

```diff
@@ -150,15 +149,20 @@ private synchronized void startConsuming(Subscriber<? super Message> subscriber,
         );
         InternalConsumer cons;
         String threadNamePrefix = "consume-thread-" + consumerCount;
-        if (consumerRef.get()==null) {
+        if (consumerRef.get() == null) {
             cons = new InternalConsumer(channel, subscriber, closeTimeout, threadNamePrefix, metricsReporter, new AtomicLong(), new AtomicLong());
-        }else{
+        } else {
             cons = new InternalConsumer(consumerRef.get(), threadNamePrefix, channel, subscriber);
         }
-        channel.basicConsume(
-                consumerTag,
-                cons);
         consumerRef.set(cons);
+        try {
+            channel.basicConsume(
+                    consumerTag,
+                    cons);
+        } catch (Exception e) {
+            cons.close();
+            throw e;
+        }
```

Comment:
cons.close is closed in onUnsubscribe method, but it's never called if Observable failed to create. There's actually a better solution, let me push another commit.